src/software/thunderscope/robot_diagnostics/chicker.py

```diff
@@ -73,6 +96,86 @@ def __init__(self, proto_unix_io):
         self.geneva_value = 3
         self.power_value = 1
 
+    def kick_clicked(self):
+        # if button is enabled
+        if self.kick_button_enable:
+            # send kick primitive
+            self.send_command(ChickerCommandMode.KICK)
+            self.toggle_kick_button()
+
+            # set and start timer to re-enable kick button after 3 seconds
+            self.start_timer_once(self.toggle_kick_button, 3000)
+
+    def chip_clicked(self):
+        # if button is enabled
+        if self.chip_button_enable:
+            # send chip primitive
+            self.send_command(ChickerCommandMode.CHIP)
+            self.toggle_chip_button()
+
+            # set and start timer to re-enable chip button after 3 seconds
+            self.start_timer_once(self.toggle_chip_button, 3000)
+
+    def start_timer_once(self, function, duration):
+        """Starts a QTimer to call the given function once after the given duration
+
+        :param function: function to call after duration
+        :param duration: duration to call function after
+        :returns: None
+
+        """
+
+        t = QTimer(self)
+        t.setTimerType(Qt.TimerType.PreciseTimer)
+        t.timeout.connect(function)
+        t.setSingleShot(True)
+        t.start(duration)
+
+    def toggle_kick_button(self):
+        self.kick_button_enable = not self.kick_button_enable
+
+    def toggle_chip_button(self):
+        self.chip_button_enable = not self.chip_button_enable
+
+    def buttons_disable(self):
+        self.kick_button_enable = False
+        self.chip_button_enable = False
+
+    def buttons_enable(self):
+        self.kick_button_enable = True
+        self.chip_button_enable = True
+
+    def send_command(self, command):
+        """Sends a kick or chip primitive dependent on boolean parameter
```

Comment:
nit: the parameter is not a boolean
```suggestion
        """Sends a (auto)kick or (auto)chip primitive
```

Reply:
done